pages/index.vue

```diff
@@ -1,19 +1,82 @@
 </template>
 
 <script lang="ts">
-import Vue from "vue";
-import VueLayerMarker from "~/components/VueLayerMarker.vue";
+    import Vue from "vue";
+    import { connect } from "mqtt";
+    import VueLayerMarker from "~/components/VueLayerMarker.vue";
 
-export default Vue.extend({
-  name: "IndexPage",
-  components: { VueLayerMarker }
-});
+    export default Vue.extend({
+        name: "IndexPage",
+        components: { VueLayerMarker },
+        data () {
+            return {
+                longitude: 0,
+                latitude: 0
+            };
+        },
+        mounted () {
+            this.connectToClient();
+        },
+        methods: {
+            connectToClient () {
+                const token = process.env.FLESPI_KEY || "not found"; // TODO : Handle when the key isn't there
+                if (token.length !== 64) {
+                    console.log("please check the token");
+                    return;
+                }
+                const client = connect("wss://mqtt.flespi.io", {
+                    clientId: "track-and-trace",
+                    // see https://flespi.com/kb/tokens-access-keys-to-flespi-platform to read about flespi tokens
+                    username: "FlespiToken " + token,
+                    protocolVersion: 5,
+                    clean: true
+                });
+
+                // TODO : we should be able to see all trackers and not only one
+                client.on("connect", () => {
+                    // Subscribe to the telemetry topic
+                    client.subscribe("flespi/state/gw/devices/4530445/telemetry/#", { qos: 1 }, (err) => {
+                        if (err) {
+                            console.log("failed to subscribe to topic \"test\":", err);
+                            return;
+                        }
+                        console.log("subscribed to \"test\"");
+                    });
+                });
+
+                client.on("message", (topic, msg) => {
+                    this.filter(topic, msg.toString("utf8"));
+                });
+
+                client.on("close", () => {
+                    console.log("disconnected");
+                });
+
+                client.on("error", (err) => {
+                    console.log("mqtt client error:", err);
+                    client.end(true); // force disconnect
+                });
+            },
+            filter (topic: string, msg: string) {
+                const splitTopic: string[] = topic.split("/");
+                const informationName: string = splitTopic[splitTopic.length - 1];
+
+                if (informationName === "position.longitude") {
+                    // ref[0] = Number(msg);
+                    this.longitude = Number(msg);
+                } else if (informationName === "position.latitude") {
+                    // ref[1] = Number(msg);
+                    this.latitude = Number(msg);
+                }
+            }
+        }
```

Comment:
This works, but I suggest moving this to a [plugin](https://nuxtjs.org/docs/directory-structure/plugins/) in order to get some more structure and clarity. Best you make an issue of this so that you don't forget it.

Reply:
Excellent point to remove all `console.log()` before deployment!
You're using ESLint ; I would add the [`no-console`](https://eslint.org/docs/latest/rules/no-console) rule so it will automatically tell you if there are `console.log()` left in your code.

Reply:
It already does that, but it's a warning by default instead of a straight error if I remember correctly